http/src/main/resources/init-resources/startup.sh

```diff
@@ -108,6 +108,8 @@ then
     if [ ! -z "$JUPYTER_DOCKER_IMAGE" ] ; then
         echo "Restarting Jupyter Container $GOOGLE_PROJECT / $CLUSTER_NAME..."
 
+        docker exec $JUPYTER_SERVER_NAME /bin/bash -c "[ -d $JUPYTER_USER_HOME/notebooks ] && [ ! -d $JUPYTER_USER_HOME/notebooks/.jupyter ] && rsync -av --progress --exclude notebooks . $JUPYTER_USER_HOME/notebooks || true"
```

Comment:
Thanks for the fix. Unfortunately I don't really understand it - why copy HOME (presumably, that's the cwd) -> notebooks ? I thought the notebooks dir went away with your changes?

Reply:
so this line is only for migration. Say you have an existing runtime, jupyter container's PD is mapped at `$HOME/notebooks`, but all jupyter related files (.jupyter, .local) and things like bash history etc all lives under `$HOME`. The home diretory change will make it so that next time this runtime starts up, PD will be mapped to `$HOME`, but this means that the previous  files under $HOME (`.jupyter`, `.local` etc) will be lost....So this one line is to before we restart jupyter container with updated home directory mapping, we will copy all files under `$HOME` to `$HOME/notebooks` first, which will live on PD now...So when it starts up, what was previously under $HOME will now appear in new $HOME as well

Reply:
I see - thanks! May be worth a comment here

Reply:
+1 on comment :)

Reply:
it's already added